### PR TITLE
[API,FIX,TEST] Remove nlohmann::json from OpenMS headers

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

Fixes the build failure reported on #10079: a Nix build of 3.6.0 dies with `ThermoRawFileMetadata.h:13:10: fatal error: nlohmann/json.hpp: No such file or directory`. `nlohmann_json::nlohmann_json` is a PRIVATE dependency of libOpenMS, but two headers under `include/` pulled it in, and `ThermoRawFileMetadata_test` (built unconditionally, linked only against libOpenMS and the test framework) compiled one of them. vcpkg CI hides this because every dependency header lives in the one install prefix that `Boost_INCLUDE_DIRS` already puts on the include path.

After this PR no header under `src/openms/include` mentions nlohmann.

### 1. Typed API for `ThermoRawFileMetadata`

- `ThermoRawFileMetadata.h` has no external includes. The JSON-typed functions became a typed API over new value types in `OpenMS::Internal`: `ThermoTrailerEntry`, `ThermoReaction`, `ThermoScan`, `ThermoPrecursor`. `number()` returns `std::optional<double>` instead of a JSON null-or-number; `precursors()` returns `std::vector<ThermoPrecursor>`. The class is `OPENMS_DLLAPI`.
- The implementation moved to `source/FORMAT/HANDLERS/ThermoRawFileMetadata.cpp`, compiled unconditionally (no bridge dependency). The logic is carried over one-to-one: filter-key regex, master-scan validation, same-order sibling handling, reaction-index fallback, supplemental detection, SPS mass parsing for legacy and modern labels, ancestor resolution through the parent chain.
- The JSON boundary stays in `ThermoRawFile.cpp`: a file-local `toThermoScan()` converts each parsed scan object (schema version 1: `scan_number`, `ms_level`, `filter`, `native_id`, `trailer[{label,value}]`, `reactions[{precursor_mass, isolation_width, isolation_offset, activation, collision_energy, collision_energy_valid}]`, nullable numbers) once, right after `Json::parse`. Trailer lookups, the activation lambda and the precursor loop read struct fields.
- The class test is rewritten on the ClassTest framework against the structs. Every previous assertion carries over; it no longer includes nlohmann and needs no extra link line.

### 2. `ProFormaDataJson.h` is private

Moved to `source/CHEMISTRY/ProFormaDataJson.h` (precedent: `ANALYSIS/ID/PercolatorImpl.h`, `FORMAT/DATAACCESS/ParquetReaderHelpers.h`), removed from the installed header list, included as `"ProFormaDataJson.h"` from `ProForma.cpp`. The stale trailing comment that referred to a non-existent `ProFormaData.h` / `ProFormaDataJson.cpp` is fixed.

### 3. Regression guard

`install_headers` in `cmake/install_macros.cmake` now scans every header it installs with `file(STRINGS ... REGEX "nlohmann/json")` and stops configuration with a `FATAL_ERROR` naming the file. Configure-time only. Header lists are relative to the calling source directory, so paths are resolved against `CMAKE_CURRENT_SOURCE_DIR` before scanning.

### Validation

- Fresh configure without vcpkg and with `-DWITH_THERMO_RAW=OFF` (the failing configuration): `ThermoRawFileMetadata_test` builds and passes. Its compile line carries no nlohmann include directory, and the preprocessed test translation unit contains zero occurrences of `nlohmann`. The second check matters because the json headers are installed system-wide on this machine, so the include-directory check alone would not prove anything.
- `WITH_THERMO_RAW=ON` (bridge 0.3.0 built from source): `ThermoRawFileMetadata_test` and `ThermoRawFile_test` pass. `FileConverter -test -RawToMzML:reader inprocess` on `Angiotensin_AllScans.raw` (1775 spectra, 3377 precursor elements, 1688 spectrum references and supplemental activations) writes an mzML that is byte-identical to the one written by current `develop` (00ed02c6ae) from the same binary options; the baseline was generated twice to confirm determinism.
- Full class-test and TOPP suites (2851 tests; pyOpenMS and tutorial tests excluded): 2843 passed, 8 failed, all pre-existing and environmental on this machine and each verified from its log: 6x `TOPP_CometAdapter_*_out1` (a Comet 2023.01 binary from an old install vs. references written with Comet 2025.01), `TOPP_PercolatorAdapter_1_subprocess_out` (external percolator q-value), `Doxygen_Warning_test` (`HAVE_DOT` not set, no Graphviz).
- Guard: with `ProFormaDataJson.h` still installed, configure stops with `Installed header include/OpenMS/CHEMISTRY/ProFormaDataJson.h includes nlohmann/json`; a temporary `#include <nlohmann/json.hpp>` appended to `ThermoRawFile.h` stops it naming that file; with both removed, configure passes. Two things the red runs caught: the header lists are relative paths, so a first version that checked `if(EXISTS)` silently scanned nothing, and a fresh tree showed that build-time-generated headers (`build_config.h`) do not exist at configure time, so the scan is limited to files already on disk.

## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmake/install_macros.cmake [hunk 1]
```diff
@@ -32,6 +32,22 @@ endmacro()
...
+    # own templates, so only what is already on disk is scanned.
+    set(_header_to_scan "${_header}")
+    if (NOT IS_ABSOLUTE "${_header_to_scan}")
+      set(_header_to_scan "${CMAKE_CURRENT_SOURCE_DIR}/${_header_to_scan}")
+    endif()
+    if (EXISTS "${_header_to_scan}" AND NOT IS_DIRECTORY "${_header_to_scan}")
+      file(STRINGS "${_header_to_scan}" _nlohmann_json_hits REGEX "nlohmann/json")
+      if (_nlohmann_json_hits)
+        message(FATAL_ERROR "Installed header ${_header} includes nlohmann/json. Keep JSON types out of "
+                            "public headers: move the header under source/ or expose value types instead.")
```
Comment (timosachsenberg) [inline cmake/install_macros.cmake:48]: @pjones should we keep/introduce such guards or do you think this just bloats the scripts?

Comment (pjones) [reply cmake/install_macros.cmake:48]: If the guard is generic so we don't have to update it all the time I think it's a nice addition.
